### PR TITLE
Edit recipe

### PR DESCRIPTION
**Pull request by ar1a**

### CODE REVIEW

**Comment by ar1a** [comment]
we will probably want to merge #10 before this

File: client/src/RecipeForm.js [hunk 1]
```diff
@@ -68,8 +70,10 @@ export default ({
       { name, Yield, ingredients: I, equipment: E },
       { setSubmitting }
     ) => {
-      const ingredients = I.map(i => omit(i, 'id'));
-      const equipment = E.map(e => omit(e, 'id'));
+      let ingredients = I.map(i => omit(i, 'id'));
+      ingredients = ingredients.map(i => omit(i, '__typename'));
+      let equipment = E.map(e => omit(e, 'id'));
+      equipment = equipment.map(e => omit(e, '__typename'));
```
Comment (delan) [inline client/src/RecipeForm.js]: np: extract these pairs of lines

File: client/src/RecipeForm.js [hunk 2]
```diff
@@ -34,6 +35,7 @@ export default ({
       ingredients,
       equipment
     }}
+    isInitialValid={!!edit}
```
Comment (delan) [inline client/src/RecipeForm.js]: np: type confusion `''` and `false` (shouldn’t matter in practice)

File: client/src/RecipeForm.js [hunk 3]
```diff
@@ -78,7 +82,11 @@ export default ({
           equipment
         });
         setSubmitting(false);
-        history.push('/'); // TODO: Redirect if edit
+        if (edit) {
+          history.push(`/recipes/${edit}`);
+        } else {
+          history.push('/'); // TODO: Redirect if edit
```
Comment (delan) [inline client/src/RecipeForm.js]: remove comment?

**Review by delan — APPROVED** [review]

**Pull request merged**